src/OrleansAWSUtils/Reminders/DynamoDBReminderTable.cs

```diff
@@ -0,0 +1,294 @@
+                    $"Intermediate error reading reminder entry {Utils.DictionaryToString(keys)} from table {TABLE_NAME_DEFAULT_VALUE}.", exc);
+                throw;
+            }
+        }
+
+        public async Task<ReminderTableData> ReadRows(GrainReference grainRef)
+        {
+            var expressionValues = new Dictionary<string, AttributeValue>
+                {
+                    { $":{SERVICE_ID_PROPERTY_NAME}", new AttributeValue(serviceId.ToString()) },
+                    { $":{GRAIN_REFERENCE_PROPERTY_NAME}", new AttributeValue(grainRef.ToKeyString()) }
+                };
+
+            try
+            {
+                var expression = $"{SERVICE_ID_PROPERTY_NAME} = :{SERVICE_ID_PROPERTY_NAME} AND {GRAIN_REFERENCE_PROPERTY_NAME} = :{GRAIN_REFERENCE_PROPERTY_NAME}";
+                var records = await storage.ScanAsync(TABLE_NAME_DEFAULT_VALUE, expressionValues, expression,
+                    item => new ReminderEntry
+                    {
+                        ETag = item[ETAG_PROPERTY_NAME].N,
+                        GrainRef = grainRef,
+                        Period = TimeSpan.Parse(item[PERIOD_PROPERTY_NAME].S),
+                        ReminderName = item[REMINDER_NAME_PROPERTY_NAME].S,
+                        StartAt = DateTime.Parse(item[START_TIME_PROPERTY_NAME].S)
+                    }).ConfigureAwait(false);
+
+                return new ReminderTableData(records);
+            }
+            catch (Exception exc)
+            {
+                logger.Warn(ErrorCode.ReminderServiceBase,
+                    $"Intermediate error reading reminder entry {Utils.DictionaryToString(expressionValues)} from table {TABLE_NAME_DEFAULT_VALUE}.", exc);
+                throw;
+            }
+        }
+
+        public async Task<ReminderTableData> ReadRows(uint beginHash, uint endHash)
+        {
+            var expressionValues = new Dictionary<string, AttributeValue>
+                {
+                    { $":{SERVICE_ID_PROPERTY_NAME}", new AttributeValue(serviceId.ToString()) },
+                    { $":Begin{GRAIN_HASH_PROPERTY_NAME}", new AttributeValue { N = beginHash.ToString() } },
+                    { $":End{GRAIN_HASH_PROPERTY_NAME}", new AttributeValue { N = endHash.ToString() } }
+                };
+
+            try
+            {
+                string expression = string.Empty;
+                if (beginHash < endHash)
+                {
+                    expression = $"{SERVICE_ID_PROPERTY_NAME} = :{SERVICE_ID_PROPERTY_NAME} AND {GRAIN_HASH_PROPERTY_NAME} > :Begin{GRAIN_HASH_PROPERTY_NAME} AND {GRAIN_HASH_PROPERTY_NAME} <= :End{GRAIN_HASH_PROPERTY_NAME}";
+                }
+                else
+                {
+                    expression = $"{SERVICE_ID_PROPERTY_NAME} = :{SERVICE_ID_PROPERTY_NAME} AND ({GRAIN_HASH_PROPERTY_NAME} > :Begin{GRAIN_HASH_PROPERTY_NAME} OR {GRAIN_HASH_PROPERTY_NAME} <= :End{GRAIN_HASH_PROPERTY_NAME})";
```

Comment:
I am not familiar enough with DynamoDB. 
Does it matter how you structure the query? I would expect it does. In Azure Table only PK and RK are sorted and indexed, so it was very important to structure the query to be done in one sorted scan.
But in DynamoDB, they have secondary indices, right?
Don't you need to tell it explicitly what columns to index? I would not index by default all columns, right?
If you do NOT index those columns, I think we need to think about efficiency of that query , same as we did for Azure Table.